### ISSUE TITLE
Implementation of JointBERT 

### ISSUE THREAD

**Issue by robertanto** [RasaHQ/rasa#9354]
Dear RASA team,

it would be very useful to have implemented the method from [BERT for Joint Intent Classification and Slot Filling](https://arxiv.org/abs/1902.10909).

Differently from the current use of BERT (as featurizer), it allows updating the architecture weights (of course, it needs a lot of data).

I see from your paper [DIET: Lightweight Language Understanding for Dialogue Systems](https://arxiv.org/pdf/2004.09936.pdf) that it outperform DIET on the ATIS and SNIPS dataset.

In general, an example of how to use HuggingFace models (all from the model hub) for joint intent and entity tasks would be a great help in developing research chatbots.

Thank you in advance!

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @hsm207 will get back to you about it soon! ✨

**Comment by hsm207** [comment]
> In general, an example of how to use HuggingFace models (all from the model hub) for joint intent and entity tasks would be a great help in developing research chatbots.

@robertanto Does the [LanguageModelFeaturizer](https://rasa.com/docs/rasa/components#languagemodelfeaturizer) component help or did you mean that you want to replace the DIETClassifier component with any model from HuggingFace's model hub?

**Comment by robertanto** [comment]
Thank you for your answer!

From the best of my knowledge, LanguageModelFeaturizer does not allow to update the weights of the pre-trained model, does it?

**Comment by hsm207** [comment]
Yes, LangugageModelFeaturizer does not update the weights of the pre-trained model.

**Comment by robertanto** [comment]
In the paper I cited in the issue, it seems that the authors trained also the BERT's weights.

**Comment by robertanto** [comment]
@hsm207, I confirm that in the paper the whole architecture is trained (starting from initial weights). 

Here an unofficial implementation: [JointBERT](https://github.com/monologg/JointBERT).

The possibility of using any transformer from HuggingFace (not only those ones in LanguageModelFeaturizer) as a backbone for a Joint model (as in the case of JointBert) would be a great feature for RASA. From this point of view, it is crucial to allow updating the backbone's weights (as I read in your documentation, this is not true for LanguageModelFeaturizer).

**Comment by hsm207** [comment]
@robertanto 

The possibility to update the backbone's weights in the LanguageModelFaturizer may be something we will consider in the future. For now, if you would like experiment replacing the DIET Classifier with another model e.g. JointBERT then you will need to implement a custom model. The Rasa NLU Examples as some examples [here](https://rasahq.github.io/rasa-nlu-examples/#intent-classifiers) to help you get started.

**Comment by robertanto** [comment]
It would be still very useful an example of a joint model based on the HuggingFace transformers library.

**Comment by hsm207** [comment]
I've let our research team know about this.
